src/storage/src/sst/parquet.rs

```diff
@@ -282,6 +297,121 @@ impl<'a> ParquetReader<'a> {
+        let ts_col_idx = self
+            .projected_schema
+            .schema_to_read()
+            .schema()
+            .timestamp_index()?;
+        let ts_col = self
+            .projected_schema
+            .schema_to_read()
+            .schema()
+            .timestamp_column()?;
+
+        let ts_col_unit = match &ts_col.data_type {
+            ConcreteDataType::Int64(_) => TimeUnit::Millisecond,
+            ConcreteDataType::Timestamp(ts_type) => ts_type.unit(),
+            _ => unreachable!(),
+        };
+
+        // build lower and upper bound according to time range and timestamp column data type.
+        let lower = self
+            .time_range
+            .start()
+            .and_then(|s| s.convert_to(ts_col_unit))
+            .map(|t| t.value())
+            .unwrap_or(i64::MIN);
+
+        let upper = self
+            .time_range
+            .end()
+            .and_then(|s| s.convert_to_ceil(ts_col_unit)) // convert to ceil to relax time range and prevent data loss caused by rounding error.
+            .map(|t| t.value())
+            .unwrap_or(i64::MAX);
+
+        let projection = ProjectionMask::roots(schema_desc, vec![ts_col_idx]);
+        let filter = RowFilter::new(vec![Box::new(TimestampRowFilter::new(
+            ts_col_idx, projection, lower, upper,
+        ))]);
+        Some(filter)
+    }
+}
+
+/// `TimestampRowFilter` is used to filter rows within given timestamp range when reading
+/// row groups from parquet files, while avoids fetching all columns from SSTs file.
+struct TimestampRowFilter {
+    timestamp_index: usize,
+    lower_bound: i64,
+    upper_bound: i64,
+    projection: ProjectionMask,
+}
+
+impl TimestampRowFilter {
+    fn new(
+        ts_col_idx: usize,
+        projection: ProjectionMask,
+        lower_bound: i64,
+        upper_bound: i64,
+    ) -> Self {
+        Self {
+            timestamp_index: ts_col_idx,
+            lower_bound,
+            upper_bound,
+            projection,
+        }
+    }
+}
+
+impl ArrowPredicate for TimestampRowFilter {
+    fn projection(&self) -> &ProjectionMask {
+        &self.projection
+    }
+
+    /// Selects the rows matching given time range.
+    fn evaluate(&mut self, batch: RecordBatch) -> std::result::Result<BooleanArray, ArrowError> {
+        let row_cnt = batch.num_rows();
+        let ts_col = batch.column(self.timestamp_index);
+
+        macro_rules! downcast_and_compute {
+            ($typ: ty) => {
+                {
+                    let ts_col = ts_col
+                        .as_any()
+                        .downcast_ref::<$typ>()
+                        .unwrap(); // safety: we've checked the data type of timestamp column.
+                    let lower_bound = PrimitiveArray::from_value(self.lower_bound, row_cnt);
+                    let upper_bound = PrimitiveArray::from_value(self.upper_bound, row_cnt);
+                    let left = arrow::compute::gt_eq(ts_col, &lower_bound)?;
```

Comment:
Could we use [gt_eq_scalar()](https://docs.rs/arrow/latest/arrow/compute/kernels/comparison/fn.gt_eq_scalar.html) instead of create a `row_cnt` size array? The same as `lt`.